pkg/azureclients/privatelinkserviceclient/azure_privatelinkserviceclient.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package privatelinkserviceclient implements the client for PrivateLinkService.
+package privatelinkserviceclient
```

Comment:
could you add some unit tests for the new clients? you could refer other clients how to add them

Reply:
I'm a little bit confused about whether we need to write unit test to invoke mock client. because actual implementation is not tested.

Reply:
That's not true, Only arm mock clients are used in the unit tests, hence the codes in this client should be validated by unit tests.

Reply:
I agree that we should add units test to cover these code. But this client object is initialized using azure sdk.  And azure sdk doesn't provide mock client. Should I rewrite the logic and use armclient instead?

Reply:
correct, please switch to armclient